src/main/java/org/springframework/data/redis/listener/RedisMessageListenerContainer.java

```diff
@@ -689,36 +690,55 @@ private class SubscriptionTask implements SchedulingAwareRunnable {
 		private class PatternSubscriptionTask implements SchedulingAwareRunnable {
 
 			private long WAIT = 500;
-			private long ROUNDS = 3;
```

Comment:
This pattern subscription thread will now run until it's canceled or if it's done.